src/Extension/DebugExtension.php

```diff
@@ -22,10 +22,8 @@ public function getFunctions(): array
     {
         // dump is safe if var_dump is overridden by xdebug
         $isDumpOutputHtmlSafe = \extension_loaded('xdebug')
-            // false means that it was not set (and the default is on) or it explicitly enabled
-            && (false === \ini_get('xdebug.overload_var_dump') || \ini_get('xdebug.overload_var_dump'))
-            // false means that it was not set (and the default is on) or it explicitly enabled
-            // xdebug.overload_var_dump produces HTML only when html_errors is also enabled
+            // Xdebug overloads var_dump in develop mode when html_errors is enabled
+            && str_contains(\ini_get('xdebug.mode'), 'develop')
```

Comment:
Note: the safer check would be:

```suggestion
            && in_array('develop', xdebug_info('mode'), true)
```

But that does not work with Xdebug 3.0 which could technically be installed in environments with PHP 8.0.